### PR TITLE
Add some convenience functions for handling properties

### PR DESCRIPTION
**Pull request by andyferris**
Export the following functions

 * `getproperties` which lets you specify a tuple of symbols to get a subset of properties from a table or row.
 * `deleteproperty` which lets you specify a symbols to get a all properties except the specified one from a table or row.
 * `deleteproperties` which lets you specify a tuple of symbols to get a all properties except the specified ones from a table or row.

CC @EvertSchippers

### CODE REVIEW

File: src/properties.jl [hunk 1]
```diff
@@ -36,7 +36,29 @@ end
 @inline GetProperties(names::Tuple{Vararg{Symbol}}) = GetProperties{names}()
 
 """
-    getproperties(names::Symbol...)
+    getproperties(object, names::Tuple{Vararg{Symbol}})
+
+Return a function that extracts a set of properties with the given `names` from an `object`,
```
Comment (EvertSchippers) [inline src/properties.jl:41]: This one does not return a function (doc string copied)

File: src/TypedTables.jl [hunk 2]
```diff
@@ -40,6 +40,10 @@ if VERSION < v"1.2.0-DEV.291"
     end
 end
 
+# Apparently by default this is slower than necessary because we go through generic introspection.
+# TODO - we should probably make a PR for Base.
+Base.propertynames(::NamedTuple{names}) where {names} = names
```
Comment (EvertSchippers) [inline src/TypedTables.jl:45]: This one is also good to have indeed!

**Review by EvertSchippers — COMMENTED** [review]
Cool!

File: src/properties.jl (hunk 1 above)
Comment (andyferris) [reply src/properties.jl:41]: Good catch - thanks

**Pull request merged**